modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * @author Eyal Wolanowski <eyalvo@matrix.co.il>
+ * FHIR ADDRESS trait
+ */
+
+namespace FhirAPI\FhirRestApiBuilder\Parts\Strategy\Traits;
+
+
+use FhirAPI\FhirRestApiBuilder\Parts\Strategy\StrategyElement\ValueSet\ValueSet;
+
+trait FHIRElementValidation
+{
+    private $valueSet = null;
+
+    /**
+     * map and run a validation function
+     *
+     * @param $validator array
+     * @param $data array
+     * @param $mainTable string
+     *
+     * @return bool
+     */
+    public function validate($validator, $data, $mainTable = null)
+    {
+        $this->valueSet = new ValueSet(array(
```

Comment:
better to init the new ValueSet in the checkIfInList method, with singlton logic (prevent unnecessary initializes)

Reply:
ok